docs/source/data-model.rst

```diff
@@ -1,18 +1,39 @@
-**********
-Data Model
-**********
+********************************
+Data Model: Documents and Events
+********************************
+
+*begin remove*
 
 Bluesky's event-based data model supports complex, asynchronous data collection
 and enables sophisticated live, prompt, streaming, and *post-facto* data
 analysis.
 
+*end remove*
+
+A primary design goal of bluesky is to enable better research by recording
+rich metadata alongside measured data for use in later analysis. Documents are
+how we do this.
+
+A *document* is our term for a Python dictionary with a schema --- that is,
+organized in a
+`formally specified <https://github.com/NSLS-II/event-model>`_ way --- created
+by the RunEngine during plan execution.  All of the metadata and data generated
+by executing the plan is organized into documents.
+
+A :doc:"later section <callbacks>" describes how outside functions can
```

Comment:
This is one of the bluesky internal references of the kind I think we should just remove. Maybe this whole paragraph can go?

Reply:
I liked the mention of subscribing to a document stream. It convinces me that this is very useful information. I propose to link to the Bluesky documentation:
```
The `bluesky documentation <https://blueskyproject.io/bluesky/>`_ describes
how outside functions can "subscribe" to a stream of these documents,
visualizing, processing, or saving them. This section provides an outline of
documents themselves, aiming to give a sense of the structure and familiarity
with useful components.
```

Reply:
That makes a lot of sense.